### ISSUE TITLE
Dataframe with Decimal Type cannot be directly saved to BQ  table

### ISSUE THREAD

**Issue by GaryLiuTelus** [GoogleCloudDataproc/spark-bigquery-connector#783]
When  saving spark dataframe to BQ table with code:
```
df.write.mode("overwrite").format("bigquery").option("writeMethod", "direct").save("<dataset>.<table>")
```
, it gave the following error if there are Decimal Type columns in spark dataframe: 

```
Py4JJavaError: An error occurred while calling o844.save.
: com.google.cloud.spark.bigquery.repackaged.com.google.inject.ProvisionException: Unable to provision, see the following errors:

1) Error in custom provider, java.lang.NullPointerException: java.lang.IllegalStateException: Unexpected type: DecimalType(x,y)
  at com.google.cloud.spark.bigquery.write.context.BigQueryDataSourceWriterModule.provideDirectDataSourceWriterContext(BigQueryDataSourceWriterModule.java:60)
  while locating com.google.cloud.spark.bigquery.write.context.BigQueryDirectDataSourceWriterContext
```

But if use indirect way to write, it is OK:

```
df.write.mode("overwrite").format("bigquery").option("temporaryGcsBucket", "<my-gcs_bucket-name>").save("<dataset>.<table>")
```

I am using `spark-bigquery-with-dependencies_2.12-0.26.0.jar`
spark version 3.2.1,
python 3.7.12
scala 2.12

**Comment by kristopherkane** [comment]
I can replicate this on Spark 3.1.3 with Scala 2.12. (Dataproc 2.0) - using version spark-bigquery-with-dependencies_2.12-0.26.0.jar

Indirect write works. 

Our Spark decimal column(10,3) maps to a BQ Numeric type. 
```

Caused by: java.lang.NullPointerException: java.lang.IllegalStateException: Unexpected type: DecimalType(10,3)
	at com.google.cloud.spark.bigquery.repackaged.com.google.common.base.Preconditions.checkNotNull(Preconditions.java:907)
	at com.google.cloud.spark.bigquery.ProtobufUtils.toProtoFieldType(ProtobufUtils.java:461)
	at com.google.cloud.spark.bigquery.ProtobufUtils.buildDescriptorProtoWithFields(ProtobufUtils.java:447)
	at com.google.cloud.spark.bigquery.ProtobufUtils.toDescriptor(ProtobufUtils.java:299)
	at com.google.cloud.spark.bigquery.ProtobufUtils.toProtoSchema(ProtobufUtils.java:136)
	at com.google.cloud.spark.bigquery.write.context.BigQueryDirectDataSourceWriterContext.<init>(BigQueryDirectDataSourceWriterContext.java:91)
	at com.google.cloud.spark.bigquery.write.context.BigQueryDataSourceWriterModule.provideDirectDataSourceWriterContext(BigQueryDataSourceWriterModule.java:72)
	at com.google.cloud.spark.bigquery.write.context.BigQueryDataSourceWriterModule$$FastClassByGuice$$c1c9fbee.invoke(<generated>)
	at com.google.cloud.spark.bigquery.repackaged.com.google.inject.internal.ProviderMethod$FastClassProviderMethod.doProvision(ProviderMethod.java:264)
	at com.google.cloud.spark.bigquery.repackaged.com.google.inject.internal.ProviderMethod.doProvision(ProviderMethod.java:173)
	at com.google.cloud.spark.bigquery.repackaged.com.google.inject.internal.InternalProviderInstanceBindingImpl$CyclicFactory.provision(InternalProviderInstanceBindingImpl.java:185)
	at com.google.cloud.spark.bigquery.repackaged.com.google.inject.internal.InternalProviderInstanceBindingImpl$CyclicFactory.get(InternalProviderInstanceBindingImpl.java:162)
	at com.google.cloud.spark.bigquery.repackaged.com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
	at com.google.cloud.spark.bigquery.repackaged.com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:168)
	at com.google.cloud.spark.bigquery.repackaged.com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:39)
	at com.google.cloud.spark.bigquery.repackaged.com.google.inject.internal.InjectorImpl$1.get(InjectorImpl.java:1094)
```

**Comment by davidrabinowitz** [comment]
Fixed in version 0.31.0
